x-pack/plugins/apm/public/components/app/service_dashboards/actions/save_dashboard_modal.tsx

```diff
@@ -0,0 +1,266 @@
+  EuiModalHeaderTitle,
+  EuiSwitch,
+  EuiModalBody,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiToolTip,
+  EuiIcon,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { DashboardItem } from '@kbn/dashboard-plugin/common/content_management';
+import { callApmApi } from '../../../../services/rest/create_call_apm_api';
+import { useDashboardFetcher } from '../../../../hooks/use_dashboards_fetcher';
+import { FETCH_STATUS } from '../../../../hooks/use_fetcher';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { DashboardTypeEnum } from '../../../../../common/service_dashboards';
+import { SavedServiceDashboard } from '../../../../../common/service_dashboards';
+
+interface Props {
+  onClose: () => void;
+  onRefresh: () => void;
+  currentDashboard?: SavedServiceDashboard;
+  serviceDashboards?: SavedServiceDashboard[];
+}
+
+export function SaveDashboardModal({
+  onClose,
+  onRefresh,
+  currentDashboard,
+  serviceDashboards,
+}: Props) {
+  const {
+    core: { notifications },
+  } = useApmPluginContext();
+  const { data: allAvailableDashboards, status } = useDashboardFetcher();
+  let defaultOption: EuiComboBoxOptionOption<string> | undefined;
+
+  const [useContextFilter, setUseContextFilter] = useState(
+    currentDashboard?.useContextFilter ?? true
+  );
+
+  if (currentDashboard) {
+    const { dashboardTitle, dashboardSavedObjectId } = currentDashboard;
+    defaultOption = { label: dashboardTitle, value: dashboardSavedObjectId };
+  }
+
+  const [selectedDashboard, setSelectedDashboard] = useState(
+    defaultOption ? [defaultOption] : []
+  );
+
+  const isEditMode = !!currentDashboard?.id;
+
+  const {
+    path: { serviceName },
+  } = useApmParams('/services/{serviceName}/dashboards');
+
+  const reloadServiceDashboards = useCallback(() => {
+    onRefresh();
+  }, [onRefresh]);
+
+  const options = allAvailableDashboards?.map(
+    (dashboardItem: DashboardItem) => ({
+      label: dashboardItem.attributes.title,
+      value: dashboardItem.id,
+      disabled:
+        serviceDashboards?.some(
+          ({ dashboardSavedObjectId }) =>
+            dashboardItem.id === dashboardSavedObjectId
+        ) ?? false,
+    })
+  );
+  const onSave = useCallback(
+    async function () {
+      const [newDashboard] = selectedDashboard;
+      try {
+        if (newDashboard.value) {
+          await callApmApi('POST /internal/apm/service-dashboard', {
+            params: {
+              query: { serviceDashboardId: currentDashboard?.id },
+              body: {
+                dashboardTitle: newDashboard.label,
+                dashboardSavedObjectId: newDashboard.value,
+                useContextFilter,
+                linkTo: DashboardTypeEnum.single, // iteration-1: Only single supported
+                serviceName,
+                kuery: undefined,
```

Comment:
I think we should need `kuery` - nothing else. In the first iteration it should always be linked to the current service.
```suggestion
                kuery: `service.name: ${currentServiceName}`,
```